spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/azure/openai/AzureOpenAiAutoConfiguration.java

```diff
@@ -36,26 +33,82 @@
 		AzureOpenAiConnectionProperties.class, AzureOpenAiImageOptionsProperties.class })
 public class AzureOpenAiAutoConfiguration {
 
+	private final static String APPLICATION_ID = "spring-ai";
+
 	@Bean
-	@ConditionalOnMissingBean
+	@ConditionalOnMissingBean({ OpenAIClient.class, TokenCredential.class })
 	public OpenAIClient openAIClient(AzureOpenAiConnectionProperties connectionProperties) {
 
-		Assert.hasText(connectionProperties.getApiKey(), "API key must not be empty");
+		if (StringUtils.hasText(connectionProperties.getApiKey())) {
+
+			Assert.hasText(connectionProperties.getEndpoint(), "Endpoint must not be empty");
+
+			return new OpenAIClientBuilder().endpoint(connectionProperties.getEndpoint())
+				.credential(new AzureKeyCredential(connectionProperties.getApiKey()))
+				.clientOptions(new ClientOptions().setApplicationId(APPLICATION_ID))
+				.buildClient();
+		}
+
+		// Connect to OpenAI (e.g. not the Azure OpenAI). The deploymentName property is
+		// used as OpenAI model name.
+		if (StringUtils.hasText(connectionProperties.getOpenAiApiKey())) {
+			return new OpenAIClientBuilder().endpoint("https://api.openai.com/v1")
+				.credential(new KeyCredential(connectionProperties.getOpenAiApiKey()))
+				.clientOptions(new ClientOptions().setApplicationId(APPLICATION_ID))
+				.buildClient();
+		}
+
+		throw new IllegalArgumentException("Either API key or OpenAI API key must not be empty");
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnBean(TokenCredential.class)
+	public OpenAIClient openAIClientWithTokenCredential(AzureOpenAiConnectionProperties connectionProperties,
+			TokenCredential tokenCredential) {
+
+		Assert.notNull(tokenCredential, "TokenCredential must not be null");
 		Assert.hasText(connectionProperties.getEndpoint(), "Endpoint must not be empty");
 
 		return new OpenAIClientBuilder().endpoint(connectionProperties.getEndpoint())
-			.credential(new AzureKeyCredential(connectionProperties.getApiKey()))
-			.clientOptions(new ClientOptions().setApplicationId("spring-ai"))
+			.credential(tokenCredential)
+			.clientOptions(new ClientOptions().setApplicationId(APPLICATION_ID))
 			.buildClient();
 	}
 
+	/**
+	 * If the environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET
+	 * are set, the DefaultAzureCredential can be used to authenticate the client.
+	 * @param clientId Azure client id
+	 * @param tenantId Azure tenant id
+	 * @param clientSecret Azure client secret
+	 * @return new TokenCredential instance
+	 */
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnProperty(name = { "azure.client.id", "azure.tenant.id", "azure.client.secret" },
+			matchIfMissing = false)
+	public TokenCredential createTokenCredential(@Value("${azure.client.id}") String clientId,
+			@Value("${azure.tenant.id}") String tenantId, @Value("${azure.client.secret}") String clientSecret) {
+		return new DefaultAzureCredentialBuilder().build();
```

Comment:
The application properties are not being used to create the TokenCredential.  The `DefaultAzureCredentialBuilder` does have a tenantId property, but does not have a property such as `secret` and there are two types of client id's so it doesn't match up.

Maybe we need to leverage spring cloud azure and it's support for microsoft entra id [here](https://learn.microsoft.com/en-us/azure/developer/java/spring-framework/authentication#other-credential-types).

Reply:
Please remove the  `createTokenCredential` and add the spring cloud azure documentation references to the Azure Chat and Embedding model documentations